### ISSUE TITLE
[Perf]: Performance Regression in SFPU-based INT32 MIN/MAX (L1 Sharded)

### ISSUE THREAD

**Issue by ign-sudha** [tenstorrent/tt-metal#44698]
### Component / Area

ops

### Issue Type (optional)

None

### Observed

This issue is observed with the following PR:
- #43989 

1) There is a significant performance regression for INT32 MIN/MAX operations when using L1 sharded tensors, specifically for certain shapes and reduction dimensions.
> **Note:** Values below are **Device FW Duration (ns)**.

#### Case 1: Shape (1, 1, 4096, 256), dim = -2

| Shape          | Dtype | Dim | Op  | New Branch (ns) | Main Branch (ns) | Diff (ns) | Diff (%) |
|----------------|-------|-----|-----|------------------|------------------|-----------|----------|
| (1,1,4096,256) | int32 | -2  | max | 51559.4          | 31761.3          | 19798.1   | 62.33%   |
| (1,1,4096,256) | int32 | -2  | min | 51541.2          | 31769.8          | 19771.4   | 62.23%   |

2) Additionally, there is an unexpected performance asymmetry between `dim=-1` and `dim=-2` reductions, even though the effective reduced dimension size is identical.

#### Case 2: Shape (1, 1, 256, 4096), dim = -1

| Shape          | Dtype | Dim | Op  | New Branch (ns) | Main Branch (ns) | Diff (ns) | Diff (%) |
|----------------|-------|-----|-----|------------------|------------------|-----------|----------|
| (1,1,256,4096) | int32 | -1  | max | 121040.4         | 120395.6         | 644.8     | 0.54%    |
| (1,1,256,4096) | int32 | -1  | min | 124377.0         | 122879.5         | 1497.5    | 1.22%    |

For example, comparing max:

- dim = -2: `31761.3 ns`  
- dim = -1: `120395.6 ns`  

This indicates that the `dim = -1` case is approximately **3.7× slower** than the `dim = -2` case for the same reduction size (256).

### Expected

- Performance for `dim=-1` and `dim=-2` should be comparable, since:
  - Reduction size is identical (256)
  - Same dtype (`int32`)
  - Similar L1 sharding configuration

- No significant regression vs main branch is expected.

### Configuration
- Dtype: int32
- Layout: TILE_LAYOUT
- Memory: L1 sharded
- Shapes:
  (1, 1, 4096, 256) → dim = -2
  (1, 1, 256, 4096) → dim = -1

### 2. Input data / link or description

```python
# SPDX-FileCopyrightText: © 2026 Tenstorrent USA, Inc.
#
# SPDX-License-Identifier: Apache-2.0

import pytest
import torch
import ttnn
from ttnn import (
    ShardSpec,
    ShardOrientation,
    MemoryConfig,
    BufferType,
    TensorMemoryLayout,
    CoreRangeSet,
    CoreRange,
    CoreCoord,
)

from tests.ttnn.utils_for_testing import assert_with_pcc, assert_equal

pytestmark = pytest.mark.use_module_device

BRANCH = "main"  # set to "main" to use main-branch assertion thresholds


def _assert_int32(torch_out, ttnn_out):
    if BRANCH == "sudha":
        assert_equal(ttnn_out, torch_out)
    else:
        assert_with_pcc(torch_out, ttnn_out, pcc=-1.0)


# ============================================================================
# Memory-config helpers
# ============================================================================

def _height_sharded(shard_h: int, shard_w: int, grid_y: int, grid_x: int) -> MemoryConfig:
    grid = CoreRangeSet({CoreRange(CoreCoord(0, 0), CoreCoord(grid_x - 1, grid_y - 1))})
    return MemoryConfig(
        memory_layout=TensorMemoryLayout.HEIGHT_SHARDED,
        buffer_type=BufferType.L1,
        shard_spec=ShardSpec(grid, [shard_h, shard_w], ShardOrientation.ROW_MAJOR),
    )


def _width_sharded(shard_h: int, shard_w: int, grid_y: int, grid_x: int) -> MemoryConfig:
    grid = CoreRangeSet({CoreRange(CoreCoord(0, 0), CoreCoord(grid_x - 1, grid_y - 1))})
    return MemoryConfig(
        memory_layout=TensorMemoryLayout.WIDTH_SHARDED,
        buffer_type=BufferType.L1,
        shard_spec=ShardSpec(grid, [shard_h, shard_w], ShardOrientation.ROW_MAJOR),
    )

# ============================================================================
# Shape (1, 1, 4096, 256)  —  Ht=128, Wt=8
# ============================================================================

_4096x256_dimw_in  = _height_sharded(shard_h=64,   shard_w=256, grid_y=8, grid_x=8)
_4096x256_dimw_out = _height_sharded(shard_h=64,   shard_w=32,  grid_y=8, grid_x=8)

_4096x256_dimh_in  = _width_sharded(shard_h=4096, shard_w=32, grid_y=1, grid_x=8)
_4096x256_dimh_out = _width_sharded(shard_h=32,   shard_w=32, grid_y=1, grid_x=8)


# ──────────────── int32 ────────────────

def test_int32_4096x256_dimh_max(device):
    torch_input = torch.randint(-50_000, 50_000, (1, 1, 4096, 256), dtype=torch.int32)
    torch_output = torch.amax(torch_input, dim=-2, keepdim=True)
    input_tensor = ttnn.from_torch(torch_input, layout=ttnn.TILE_LAYOUT, device=device, dtype=ttnn.int32, memory_config=_4096x256_dimh_in)
    output_tensor = ttnn.max(input_tensor, dim=-2, keepdim=True, memory_config=_4096x256_dimh_out)
    _assert_int32(torch_output, ttnn.to_torch(output_tensor))


def test_int32_4096x256_dimh_min(device):
    torch_input = torch.randint(-50_000, 50_000, (1, 1, 4096, 256), dtype=torch.int32)
    torch_output = torch.amin(torch_input, dim=-2, keepdim=True)
    input_tensor = ttnn.from_torch(torch_input, layout=ttnn.TILE_LAYOUT, device=device, dtype=ttnn.int32, memory_config=_4096x256_dimh_in)
    output_tensor = ttnn.min(input_tensor, dim=-2, keepdim=True, memory_config=_4096x256_dimh_out)
    _assert_int32(torch_output, ttnn.to_torch(output_tensor))


# ============================================================================
# Shape (1, 1, 256, 4096)  —  Ht=8, Wt=128
# ============================================================================

_256x4096_dimw_in  = _height_sharded(shard_h=32,  shard_w=4096, grid_y=1, grid_x=8)
_256x4096_dimw_out = _height_sharded(shard_h=32,  shard_w=32,   grid_y=1, grid_x=8)

_256x4096_dimh_in  = _width_sharded(shard_h=256, shard_w=64, grid_y=8, grid_x=8)
_256x4096_dimh_out = _width_sharded(shard_h=32,  shard_w=64, grid_y=8, grid_x=8)


# ──────────────── int32 ────────────────

def test_int32_256x4096_dimw_max(device):
    torch_input = torch.randint(-50_000, 50_000, (1, 1, 256, 4096), dtype=torch.int32)
    torch_output = torch.amax(torch_input, dim=-1, keepdim=True)
    input_tensor = ttnn.from_torch(torch_input, layout=ttnn.TILE_LAYOUT, device=device, dtype=ttnn.int32, memory_config=_256x4096_dimw_in)
    output_tensor = ttnn.max(input_tensor, dim=-1, keepdim=True, memory_config=_256x4096_dimw_out)
    _assert_int32(torch_output, ttnn.to_torch(output_tensor))


def test_int32_256x4096_dimw_min(device):
    torch_input = torch.randint(-50_000, 50_000, (1, 1, 256, 4096), dtype=torch.int32)
    torch_output = torch.amin(torch_input, dim=-1, keepdim=True)
    input_tensor = ttnn.from_torch(torch_input, layout=ttnn.TILE_LAYOUT, device=device, dtype=ttnn.int32, memory_config=_256x4096_dimw_in)
    output_tensor = ttnn.min(input_tensor, dim=-1, keepdim=True, memory_config=_256x4096_dimw_out)
    _assert_int32(torch_output, ttnn.to_torch(output_tensor))

```

### 3. Frequency

Always

### 1. Software Versions

Ubuntu 22.04

### 2. Hardware Details

Wormhole(`N300`)

### Is this a regression?

Unknown

### Priority

None

**Comment by ign-sampanna** [comment]
This PR to be reviewed after =>  [Bug Fix] Drive Int32 min through the LLK MIN reduce instead of -max(-x)- [#49589](https://github.com/tenstorrent/tt-metal/pull/49589)
{Due to dependency}
